src/components/ColorPicker/CustomColorPicker.tsx

```diff
@@ -32,21 +30,25 @@ export const CustomColorPicker: FC<Omit<ColorPickerProps, "palette">> = ({
     onSelect,
 }) => {
     const colorFormats = Object.values(ColorFormat).map((id) => ({ id, title: id.toLocaleUpperCase() }));
-    const [{ hsl, hsv, rgb, hex }, setColor] = useState(transformColor(currentColor));
+    const [color, setColor] = useState<Color>(currentColor);
+    const parsedColor = tinycolor(color);
+    const { r, g, b, a } = parsedColor.toRgb();
 
     useEffect(() => {
-        setColor(transformColor(currentColor));
+        setColor(currentColor);
     }, [currentColor]);
 
-    const [hexInput, setHexInput] = useState(hex.substring(1));
+    const [hexInput, setHexInput] = useState(parsedColor.toHex());
 
     useEffect(() => {
-        setHexInput(hex.substring(1));
-    }, [hex]);
+        setHexInput(parsedColor.toHex());
+    }, [color]);
 
     const handleHexChange = () => {
-        if (isValidHex(hexInput)) {
-            onSelect(toColor(currentColor, { hex: hexInput }));
+        const parsedHex = tinycolor(hexInput);
+        parsedHex.isValid();
```

Comment:
This line is duplicated and can be removed

Reply:
Removed